bahmnicore-api/src/test/java/org/bahmni/module/bahmnicore/dao/impl/BahmniPatientDaoImplLuceneIT.java

```diff
@@ -134,83 +136,180 @@ public void shouldReturnAddressAndPatientAttributes() throws Exception{
-        
+
         assertEquals(1, patients.size());
         assertEquals("SEV51'0003", response.getIdentifier());
     }
-    
+
     @Test
     public void shouldSearchPatientsByPatientIdentifierWhenThereAreMultipleSinglesInSearchString() throws Exception {
-        
+
         List<PatientResponse> patients = patientDao.getPatientsUsingLuceneSearch("'''", "", "", null, null, 100, 0, null,null, null,null,null, "c36006e5-9fbb-4f20-866b-0ece245615a1", false, false);
-        
+
         assertEquals(0, patients.size());
     }
-    
+
     @Test
     public void shouldNotReturnDuplicatePatientsEvenIfThereAreMultipleVisitsForThePatients() {
         List<PatientResponse> patients = patientDao.getPatientsUsingLuceneSearch("HOS1225", "", null, "city_village", "", 100, 0, null,"",null,null,null, "8d6c993e-c2cc-11de-8d34-0010c6affd0f", false, false);
-        
+
         assertEquals(1, patients.size());
         PatientResponse patient1 = patients.get(0);
-        
+
         assertEquals("1058GivenName", patient1.getGivenName());
     }
-    
+
     @Test
     public void shouldNotSearchExtraIdentifiersIfFilterOnAllIdenfiersIsFalse() {
         List<PatientResponse> patients = patientDao.getPatientsUsingLuceneSearch("100010", "", null, "city_village", "", 100, 0, null,"", null, null, null, "c36006e5-9fbb-4f20-866b-0ece245615a1", false, false);
-        
+
         assertEquals(0, patients.size());
     }
-    
+
     @Test
     public void shouldSearchAllIdentifiersIfFilterOnAllIdentifiersIsTrue() {
         List<PatientResponse> patients = patientDao.getPatientsUsingLuceneSearch("0001", "", null, "city_village", "", 100, 0, null,"", null, null, null, "c36006e5-9fbb-4f20-866b-0ece245615a1", false, true);
-        
+
         assertEquals(3, patients.size());
         assertEquals("{\"National ID\" : \"NAT100010\"}", patients.get(0).getExtraIdentifiers());
         assertEquals("GAN300001",patients.get(1).getIdentifier());
     }
-    
+
     @Test
     public void shouldNotReturnPatientsIfFilterOnAllIdenfiersIsTrueButNotAnExtraIdentifier() {
         List<PatientResponse> patients = patientDao.getPatientsUsingLuceneSearch("DLF200001", "", null, "city_village", "", 100, 0, null,"", null, null, null, "c36006e5-9fbb-4f20-866b-0ece245615a1", false, true);
-        
+
         assertEquals(0, patients.size());
     }
-    
+
     @Test
     public void shouldNotReturnDuplicatePatientsEvenIfTwoIdentifiersMatches() {
         List<PatientResponse> patients = patientDao.getPatientsUsingLuceneSearch("200006", "", null, "city_village", "", 100, 0, null,"", null, null, null, "c36006e5-9fbb-4f20-866b-0ece245615a1", false, true);
-        
+
         assertEquals(1, patients.size());
         PatientResponse patient = patients.get(0);
         assertTrue(patient.getIdentifier().contains("200006"));
         assertTrue(patient.getExtraIdentifiers().contains("200006"));
     }
 
-}
+    @Test
+    public void shouldSearchByName() {
```

Comment:
Surely there were already tests validating that one could search by name, could you explain here what this is in regards to existing tests?

Reply:
Or perhaps are you re-writing existing tests, but this time going through the Lucene branch of the code? I guess my remarks hold for the next few test methods as well.

Reply:
Yes. This is through the lucene branch of the code as it is